src/core/thread/mle_router.cpp

```diff
@@ -3318,7 +3318,9 @@ void MleRouter::RemoveRouterLink(Router &aRouter)
 
 void MleRouter::RemoveNeighbor(Neighbor &aNeighbor)
 {
-    if (&aNeighbor == &mParent)
+    VerifyOrExit(!aNeighbor.IsStateInvalid());
+
+    if (&aNeighbor == &mParent || &aNeighbor == &mParentCandidate)
```

Comment:
It is possible that `aNeighbor` is `mParentCandidate` here when called from `MleRouter::HandleAdvertisement`: 
https://github.com/openthread/openthread/blob/af4837d122c9da0eb6dcb9f2a1b19b1e37b738cc/src/core/thread/mle_router.cpp#L1425-L1429

Handle it same as the case `&aNeighbor == &mParent`.

Reply:
Verified together with @librasungirl that this situation could happen in which a Router is attaching to a better parent (REED), but the REED changed it's RLOC16 during attachment and the REED also sent `MLE Advertisement`.